modules/gapi/samples/face_detection.cpp

```diff
@@ -0,0 +1,758 @@
+    auto scores = outputs.at("prob1");
+    return std::make_tuple(regressions, scores);
+}
+
+//Operator fot PNet network package creation in the loop
+inline cv::gapi::GNetPackage& operator += (cv::gapi::GNetPackage& lhs, const cv::gapi::GNetPackage& rhs) {
+    lhs.networks.reserve(lhs.networks.size() + rhs.networks.size());
+    lhs.networks.insert(lhs.networks.end(), rhs.networks.begin(), rhs.networks.end());
+    return lhs;
+}
+
+static inline std::string get_pnet_level_name(const cv::Size &in_size) {
+    return "MTCNNProposal_" + std::to_string(in_size.width) + "x" + std::to_string(in_size.height);
+}
+
+int calculate_scales(const cv::Size &input_size, std::vector<double> &out_scales, std::vector<cv::Size> &out_sizes ) {
+    //calculate multi - scale and limit the maxinum side to 1000
+    //pr_scale: limit the maxinum side to 1000, < 1.0
+    double pr_scale = 1.0;
+    double h = static_cast<double>(input_size.height);
+    double w = static_cast<double>(input_size.width);
+    if (std::min(w, h) > 1000)
+    {
+        pr_scale = 1000.0 / std::min(h, w);
+        w = w * pr_scale;
+        h = h * pr_scale;
+    }
+    else if (std::max(w, h) < 1000)
+    {
+        w = w * pr_scale;
+        h = h * pr_scale;
+    }
+    //multi - scale
+    out_scales.clear();
+    out_sizes.clear();
+    const double factor = 0.709;
+    int factor_count = 0;
+    double minl = std::min(h, w);
+    while (minl >= 12)
+    {
+        const double current_scale = pr_scale * std::pow(factor, factor_count);
+        cv::Size current_size(static_cast<int>(static_cast<double>(input_size.width) * current_scale),
+                              static_cast<int>(static_cast<double>(input_size.height) * current_scale));
+        std::cout << "current_scale " << current_scale << std::endl;
+        std::cout << "current_size " << current_size << std::endl;
+        out_scales.push_back(current_scale);
+        out_sizes.push_back(current_size);
+        minl *= factor;
+        factor_count += 1;
+    }
+    std::cout << "factor_count " << factor_count << std::endl;
+    return factor_count;
+}
+
+int calculate_half_scales(const cv::Size &input_size, std::vector<double>& out_scales, std::vector<cv::Size>& out_sizes) {
+    double pr_scale = 0.5;
+    const double h = static_cast<double>(input_size.height);
+    const double w = static_cast<double>(input_size.width);
+    //multi - scale
+    out_scales.clear();
+    out_sizes.clear();
+    const double factor = 0.5;
+    int factor_count = 0;
+    double minl = std::min(h, w);
+    while (minl >= 12.0*2.0)
+    {
+        const double current_scale = pr_scale;
+        cv::Size current_size(static_cast<int>(static_cast<double>(input_size.width) * current_scale),
+                              static_cast<int>(static_cast<double>(input_size.height) * current_scale));
+        std::cout << "current_scale " << current_scale << std::endl;
+        std::cout << "current_size " << current_size << std::endl;
```

Comment:
Still not sure if input like this is required. Is it formatted well to be present in console every time the app runs?

Reply:
Removed